File: survey-sharing-frontend/src/app/entities/statistics.ts
import { Survey } from "./survey";

export class Statistics{
  id!: string;
  survey!: string;
  numberOfAnswers!: number;
  percentOfUsersWhoAnswered!: number;
  minimumAge!: number;
  maximumAge!: number;
  averageAge!: number;
  numberOfMaleUsersWhoAnswered!: number;
  numberOfFemaleUsersWhoAnswered!: number;
  numberOfInvitationsSent!: number;
  percentOfInvitationsAccepted!: number;
  numberOfDifferentCountries!: number;
  listOfFeedbacks!: string[];
  numberOfPositiveFeedbacks!: number;
  numberOfNegativeFeedbacks!: number;
  averageRating!: number;
}

File: survey_sharing_backend/src/main/java/it/lorenzobloise/survey_sharing_backend/entities/Statistics.java
package it.lorenzobloise.survey_sharing_backend.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Version;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Document("statistics")
@ToString
@EqualsAndHashCode
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Statistics {

    @Id
    private String id;
    @Indexed(unique = true)
    @JsonIgnore
    private String survey;
    private int numberOfAnswers;
    private double percentOfUsersWhoAnswered;
    private int minimumAge;
    private int maximumAge;
    private double averageAge;
    private int numberOfMaleUsersWhoAnswered;
    private int numberOfFemaleUsersWhoAnswered;
    private int numberOfInvitationsSent;
    private double percentOfInvitationsAccepted;
    private int numberOfDifferentCountries;
    private List<String> listOfFeedbacks;
    private int numberOfPositiveFeedbacks;
    private int numberOfNegativeFeedbacks;
    private double averageRating;
    @JsonIgnore
    @Version
    private Long version;

}
